lib/services/workflow-api-service.js

```diff
@@ -143,11 +143,13 @@ function workflowApiServiceFactory(
             return taskGraphProtocol.cancelTaskGraph(instanceId);
         })
         .then(function(cancelledGraphId) {
-            if (cancelledGraphId !== instanceId) {
-                // Cancelled graph ID doesn't match the requested ID.
-                throw new Errors.TaskCancellationError('Failed to cancel ' + graphId);
-            }
-            return waterline.graphobjects.needByIdentifier(graphId);
+            return waterline.graphobjects.needByIdentifier(graphId)
+            .then (function(workflow) {
+                if (workflow.active()) {
+                    throw new Errors.TaskCancellationError('Failed to cancel ' + graphId);
+                }
+                return waterline.graphobjects.needByIdentifier(graphId);
```

Comment:
This API call isn't really meant to be deterministic either, but rather something that gets kicked off into the background. An upstream caller, if they wanted to verify cancellation, would probably be better off polling the workflow object by ID after issuing the cancellation action to make sure the status changes within a certain time window.